15.md

```diff
@@ -243,6 +247,110 @@ Create a customized user experience using the `naddr` from [NIP-19](https://gith
+
+### Event `1022`: Bid confirmation
+
+```json
+{
+    "content": <json>
+    "tags": [["e" <event ID of the bid being confirmed>], ["e", <event ID of the auction>]],
+}
+```
+
+Bids should be confirmed by the merchant before being considered as valid by other clients. So clients should subscribe to *bid confirmation* events (kind `1022`) for every auction that they follow, in addition to the actual bids and should check that the pubkey of the bid confirmation matches the pubkey of the merchant (in addition to checking the signature).
+
+The `content` field is a JSON which includes *at least* a `status`, which is one of `accepted`, `rejected`, `pending`, or `winner`. `winner` is how the *winning bid* is replied to after the auction ends and the winning bid is picked by the merchant.
+
+The reasons for which a bid can be marked as `rejected` or `pending` are up to the merchant's implementation and configuration - they could be anything from basic validation errors (amount too low) to the bidder being blacklisted or to the bidder lacking sufficient *trust*, which could lead to the bid being marked as `pending` until sufficient verification is performed. The difference between the two is that `pending` bids *might* get approved after additional steps are taken by the bidder, whereas `rejected` bids can not be later approved.
+
+An additional `message` field can appear in the `content` JSON to give further context as of why a bid is `rejected` or `pending`.
+
+Another thing that can happen is - if bids happen very close to the end date of the auction - for the merchant to decide to extend the auction duration for a few more minutes. This is done by passing a `duration_extended` field as part of a bid confirmation, which would contain a number of seconds by which the initial duration is extended. So the actual end date of an auction is always `start_date + duration + (SUM(c.duration_extended) FOR c in all confirmations`.
+
+Auctions can also have a "reserve bid" - which is an amount unknown to anyone else (therefore not part of the auction event and remembered by the merchant admin software) - and represents the minimum amount for which the seller is willing to sell the item. Anything less than that and the item will not be sold, even if the auction ends and there is a top bidder! The marketplace client software has no knowledge of this amount, but it can be notified if this amount is reached or not, by passing a boolean `reserve_bid_reached` as part of the bid confirmation content JSON. Note that this is only to help the client's UI display that this amount has been reached or not (meaning there will be no winner even after the auction ends) but it does not affect the outcome of an auction, which is determined by the merchant and communicated using the `status`: `winner` described above!
```

Comment:
"reserve bid" is pretty standard in auctions. Sellers actually want to have that because ... if they would set very high "start bid", they might not get any bids at all (buyers get scared away).

But I am happy to remove these and reduce NIP complexity. In fact the implementation can work either way - having these in the NIP or not. It's still up to the merchant how the winner is picked. So yes, let's keep it simple!